django/db/models/functions/json.py

```diff
@@ -5,6 +5,69 @@
 from django.db.models.functions import Cast
 
 
+class JSONArray(Func):
+    function = "JSON_ARRAY"
+    output_field = JSONField()
+
+    def as_sql(self, compiler, connection, **extra_context):
+        if not connection.features.has_json_object_function:
+            raise NotSupportedError(
+                "JSONArray() is not supported on this database backend."
+            )
```

Comment:
It is `JSONArray` not `JSONObject`. IMO we don't need a separate feature flag for this, we can reuse `supports_json_field`.
```suggestion
        if not connection.features.supports_json_field:
            raise NotSupportedError(
                "JSONFields are not supported on this database backend."
            )
```

Reply:
A lot of boilerplate was copy/pasted from JSONObject(), obviously.